HighwayPatrol.cpp

```diff
@@ -27,11 +31,43 @@ void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
     std::cout << "\n\n";
     std::cout << name << ": vehicle is traveling " << v->speed - h->speedLimit << " miles per hour over the speed limit" << std::endl;
     if( willArrest )
-    {
-        assert(false);
+    {    std::string carType = "";
+        if(auto* c = dynamic_cast<Car*>(v))
+        {
+            carType = " car ";
+        }
+        else if(auto* d = dynamic_cast<Motorcycle*>(v))
+        {
+            carType = " moto ";
+        }
+        else if(auto* hw = dynamic_cast<HighwayPatrol*>(v))
+        {
+            carType = " highway patrolcar ";
+        }
+        else if(auto* st = dynamic_cast<SemiTruck*>(v))    
+        {
+            carType = " semitruck ";
+        }
         //print the vehicle type in this std::cout between "THE [" and "] PULL". 
-        std::cout << name << ": YOU IN THE [ " << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+        std::cout << name << ": YOU IN THE [ " << carType << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
         std::cout << "EVERYONE ELSE, SLOW DOWN!! \n\n\n";
         h->removeVehicle(v);
     }
 }
+
+void HighwayPatrol::setSpeed(int s)
+{
+    std::cout << "weewooweewooweewoo" << std::endl;
+    Vehicle::setSpeed(s);
+}
+    
+void HighwayPatrol::tryToEvade()
+{
+    
+}
```

Comment:
Remove this function.  It does not need to be overridden and your implementation does nothing
- [ ] revise